backend/app/use_case/recommendation.py

```diff
@@ -10,40 +10,91 @@
 openai.api_base = "https://api.openai.iniad.org/api/v1"
 
 
+def extract_json(openai_response: str) -> str:
+    """
+    Extract json from openai response
+    """
+    json_start_pos = openai_response.find("{")
+    json_end_pos = openai_response.rfind("}")
+    json = openai_response[json_start_pos : json_end_pos + 1]
+    return json
+
+
 def generate_recommendation_structured_format_query(query: StructuredQuery) -> str:
     place = query.place
     duration = query.duration
+    people_num = query.people_num
     budget = query.budget
     trip_style = query.trip_style
     interests = query.interests
-    foods = query.foods
+    trip_type = query.trip_type
+    remarks = query.remarks
 
-    prompt = f"Recommend itinerary in {place} for {duration} days. "
+    prompt = f"""
+#目的
+あなたには私の旅行プランナーとして、下記の要望と制約条件に従って入力された観光地における観光プランを作成してください。
+#場所
+{place}
+#期間
+{duration}日間
```

Comment:
Now the frontend receive `dateFrom` and `dateTo` and will pass to backend. So, there are 2 ways of doing this which is create a validation function for converting date to duration. Other way which I think is better is to pass the `dateFrom` and `dateTo` directly. 

Such as
```
#期間
{date_from}から{date_to}
```

So the `"date"` in the return json value can return real date instead of day 1, day 2, etc.